src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -155,26 +156,17 @@ public AgentSettings LoadSettings()
             int poolId = 0;
             string agentName = null;
             WriteSection(StringUtil.Loc("RegisterAgentSectionHeader"));
-            while (true)
-            {
-                poolName = command.GetPool();
-                try
-                {
-                    poolId = await GetPoolId(poolName);
-                }
-                catch (Exception e) when (!command.Unattended)
-                {
-                    _term.WriteError(e);
-                }
 
-                if (poolId > 0)
-                {
-                    break;
-                }
-
-                _term.WriteError(StringUtil.Loc("FailedToFindPool"));
+            Type agentType = command.DeploymentAgent ? typeof(DeploymentAgentConfiguration) : typeof(AutomationAgentConfiguration);
+            var extensionManager = HostContext.GetService<IExtensionManager>();
+            IConfigurationProvider agentProvider = (extensionManager.GetExtensions<IConfigurationProvider>()).FirstOrDefault(x => x.GetType() == agentType);
```

Comment:
You can add a field to IConfigurationProvider, like string ConfigurationType = "Build"/"Release", then compare the string instead of Type.